examples/camera-app/linux/src/clusters/camera-avstream-mgmt/camera-av-stream-manager.cpp

```diff
@@ -49,20 +49,23 @@ Protocols::InteractionModel::Status CameraAVStreamManager::VideoStreamAllocate(c
 
     for (VideoStream & stream : mCameraDeviceHAL->GetAvailableVideoStreams())
     {
-        if (!stream.isAllocated)
+        if (stream.IsCompatible(allocateArgs))
         {
             foundAvailableStream = true;
```

Comment:
Why is this variable here?  We always return after setting it to true...

Reply:
With the logic change for stream re-use, this, actually, is not the right variable name given we are using this to return ResourceExhausted error. Renamed it to track if all streams are allocated for returning ResourceExhausted error.